synapsemonitor/monitor.py

```diff
@@ -113,6 +113,47 @@ def _find_modified_entities_file(syn: Synapse, syn_id: str, days: int = 1) -> li
     return []
 
 
+def _traverse(
+    syn: Synapse,
+    synid_root: str,
+    include_types: typing.List = ["folder", "file", "project"],
+) -> list:
+    """Traverse Synapse entity hierarchy to gather all descendant
+    entities of a root entity.
+    Args:
+        syn: Synapse connection
+        synid_root: Synapse ID of root entity.
+        include_types: Must be a list of entity types (ie. [“folder”,”file”])
+            which can be found here:
+            http://docs.synapse.org/rest/org/sagebionetworks/repo/model/EntityType.html
+    Returns:
+        List of descendant Synapse IDs and root Synapse ID
+    """
+
+    synid_desc = []
+
+    # full traverse depends on examining folder entities, even if not requested
+    include_types_mod = set(include_types)
+    include_types_mod.add("folder")
+    include_types_mod = list(include_types_mod)
+
+    synid_children = syn.getChildren(parent=synid_root, includeTypes=include_types_mod)
+    for synid_child in synid_children:
+        synid_desc.extend(
+            _traverse(
```

Comment:
I just remembered, in Python there is a max recursion limit - limit of calling a function recursively ~1000 times.  So you may get into trouble here because if you do a getChildren on a folder with 1000 files, you're actually calling `_traverse` 1000 times.  We may want to do something similar as `synapseutils.walk` where we only call `_traverse` when it is a container.

Reply:
Added a check for the entity type of the children to limit recursive calls.  Now, if a child entity is neither a folder nor a project, the child entity type is added directly to the descendant list after checking that it is a requested entity type.